samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -108,14 +349,121 @@ public void start() throws SamzaException {
     // Stop consumers
     this.systemConsumers.values().forEach(systemConsumer -> systemConsumer.stop());
 
+    // Now re-create persistent stores in read-write mode, leave non-persistent stores as-is
+    createTaskStores(this.containerModel, jobContext, containerContext, storageEngineFactories, changelogSystemStreams,
+        serdes, taskInstanceMetrics, taskInstanceCollectors, StorageEngineFactory.StoreMode.ReadWrite);
+
     LOG.info("Restore complete");
   }
 
+  /**
+   * Get the {@link StorageEngine} instance with a given name for a given task.
+   * @param taskName The task name for which the storage engine is desired.
+   * @param storeName The desired store's name.
+   * @return The desired task store.
+   */
+  public Optional<StorageEngine> getStore(TaskName taskName, String storeName) {
+    return Optional.ofNullable(this.taskStores.get(taskName).get(storeName));
+  }
+
+  /**
+   *  Get all {@link StorageEngine} instance used by a given task.
+   */
+  public Map<String, StorageEngine> getAllStores(TaskName taskName) {
+    return this.taskStores.get(taskName);
+  }
+
+  /**
+   * Set of directory paths for all stores restored by this {@link ContainerStorageManager}.
+   */
+  public Set<Path> getStoreDirectoryPaths() {
+    return this.storeDirectoryPaths;
+  }
+
+  /**
+   * Obtain the base directory used by this {@link ContainerStorageManager} for all non-logged stores.
+   * If a base-directory was NOT explicitly provided during instantiation or in config, a default base directory is
+   * returned.
+   */
+  public final File getNonLoggedStorageBaseDir() {
+
+    // if a loggedStoreBaseDirectory is explicitly specified then use that
+    if (nonLoggedStoreBaseDirectory.isPresent()) {
+      return nonLoggedStoreBaseDirectory.get();
+    }
+
+    JobConfig jobConfig = new JobConfig(this.config);
+
+    if (jobConfig.getNonLoggedStorePath().isDefined()) {
+      return new File(jobConfig.getNonLoggedStorePath().get());
+    } else {
+      return getDefaultStoreBaseDir();
+    }
+  }
+
+  /**
+   * Obtain the base directory used by this {@link ContainerStorageManager} for all change-logged stores.
+   * If a base-directory was NOT explicitly provided during instantiation or in config or via an environment variable
+   * (see ShellCommandConfig.ENV_LOGGED_STORE_BASE_DIR), then a default base directory is returned.
+   */
+  public final File getLoggedStorageBaseDir() {
```

Comment:
move this logic back to its original place SamzaContainer. this ensures that all logic to resolve store paths stays in a single place. We should also take this as an opportunity to close SAMZA-1701 as a won't fix

Reply:
done